sdk/identity/identity/Troubleshooting.md

```diff
@@ -58,43 +95,30 @@ azure:identity:info EnvironmentCredential => Invoking ClientSecretCredential wit
+|`RestError` raised from the client with a status code of 401 or 403|Authentication succeeded but the authorizing Azure service responded with a 401 (Authenticate), or 403 (Forbidden) status code. This can often be caused by the `DefaultAzureCredential` authenticating an account other than the intended or that the required role assignment is not configured.|<ul><li>[Enable logging](#enabling-and-configuring-logging) to determine which credential in the chain returned the authenticating token.</li><li>In the case a credential other than the expected is returning a token, bypass this by either signing out of the corresponding development tool, or excluding the credential with the ExcludeXXXCredential property in the `DefaultAzureCredentialOptions`</li><li>Confirm that the correct RBAC role is assigned to the identity being used to authenticate. For example, the resource specific role rather than just the inherited "Owner" role.</li></ul>|
 
-| Variable name                   | Value                                                                                       |
-| ----------------------------- | ------------------------------------------------------------------------------------------- |
-| `AZURE_CLIENT_ID`               | ID of an Azure AD app.                                                                      |
-| `AZURE_TENANT_ID`               | ID of the app's Azure AD tenant.                                                            |
-| `AZURE_CLIENT_CERTIFICATE_PATH` | Path to a PEM-encoded certificate file including private key (without password protection). |
+## Troubleshoot environment credential authentication issues
 
-##### Username and password
+### Credential unavailable error
 
-| Variable name     | Value                                           |
-| ----------------- | ----------------------------------------------- |
-| `AZURE_CLIENT_ID` | ID of an Azure AD app.                          |
-| `AZURE_USERNAME`  | A username (usually an email address).          |
-| `AZURE_PASSWORD`  | The associated password for the given username. |
+| Error Message |Description| Mitigation |
+|---|---|---|
+|Environment variables aren't fully configured.|A valid combination of environment variables wasn't set.|Ensure the appropriate environment variables are set **prior to application startup** for the intended authentication method.<p/>  <ul><li>To authenticate a service principal using a client secret, ensure the variables `AZURE_CLIENT_ID`, `AZURE_TENANT_ID` and `AZURE_CLIENT_SECRET` are properly set.</li><li>To authenticate a service principal using a certificate, ensure the variables `AZURE_CLIENT_ID`, `AZURE_TENANT_ID` and `AZURE_CLIENT_CERTIFICATE_PATH` are properly set.</li><li>To authenticate a user using a password, ensure the variables `AZURE_USERNAME` and `AZURE_PASSWORD` are properly set.</li><ul>|
```

Comment:
Environment variables are not the only reason why CredentialUnavailableError could trigger. It can also trigger if the authentication endpoints (like the IMDS endpoint) is unavailable.

Reply:
Thta happens in Managed Identity cred right? We have mentioned in the `Azure Virtual Machine Managed Identity` section

Reply:
ok thanks!